source/extensions/resource_monitors/cpu_utilization/cpu_utilization_monitor.cc

```diff
@@ -0,0 +1,70 @@
+#include "source/extensions/resource_monitors/cpu_utilization/cpu_utilization_monitor.h"
+
+#include <dirent.h>
+#include <unistd.h>
+
+#include <array>
+#include <cstdlib>
+#include <fstream>
+#include <iostream>
+#include <sstream>
+#include <string>
+#include <thread>
+
+#include "absl/strings/str_split.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace ResourceMonitors {
+namespace CpuUtilizationMonitor {
+
+CpuUtilizationMonitor::CpuUtilizationMonitor(
+    const envoy::extensions::resource_monitors::cpu_utilization::v3::
+        CpuUtilizationConfig& /*config*/,
+    std::unique_ptr<CpuStatsReader> cpu_stats_reader)
+    : utilization_(0.0), cpu_stats_reader_(std::move(cpu_stats_reader)) {
+  previous_cpu_times_ = cpu_stats_reader_->getCpuTimes();
+}
+
+void CpuUtilizationMonitor::updateResourceUsage(Server::ResourceUpdateCallbacks& callbacks) {
+  CpuTimes cpu_times = cpu_stats_reader_->getCpuTimes();
+  if (!cpu_times.is_valid) {
+    ENVOY_LOG_MISC(error, "Can't open file to read CPU utilization");
```

Comment:
I think just putting the rrors in the onFailure call should be sufficient enough

Reply:
Yeah, makes sense. Removed.